foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-StoragePolicySetDefinition.json

```diff
@@ -56,6 +56,795 @@
+                            },
+                            "then": {
+                                "effect": "[[parameters('effect')]"
+                            }
+                        }
+                    },
+                    "name": "Deny-Storage-NetworkAclsVirtualNetworkRules"
+                },
+                {
+                    "properties": {
+                        "displayName": "Cross-tenant private endpoints should be restricted for Storage Accounts",
+                        "mode": "Indexed",
+                        "description": "Azure Storage accounts should restrict cross-tenant private endpoints. Enforce this for increased data exfiltration protection.",
+                        "metadata": {
+                            "version": "1.0.0",
+                            "category": "Storage"
+                        },
+                        "parameters": {
+                            "effect": {
+                                "type": "String",
+                                "metadata": {
+                                    "displayName": "Effect",
+                                    "description": "Enable or disable the execution of the policy"
+                                },
+                                "allowedValues": [
+                                    "Deny",
+                                    "Audit",
+                                    "Disabled"
+                                ],
+                                "defaultValue": "Deny"
+                            }
+                        },
+                        "policyRule": {
+                            "if": {
+                                "allOf": [
+                                    {
+                                        "field": "type",
+                                        "equals": "Microsoft.Storage/storageAccounts/privateEndpointConnections"
+                                    },
+                                    {
+                                        "field": "Microsoft.Storage/storageAccounts/privateEndpointConnections/privateLinkServiceConnectionState.status",
+                                        "equals": "Approved"
+                                    },
+                                    {
+                                        "anyOf": [
+                                            {
+                                                "field": "Microsoft.Storage/storageAccounts/privateEndpointConnections/privateEndpoint.id",
+                                                "exists": false
+                                            },
+                                            {
+                                                "value": "[[split(concat(field('Microsoft.Storage/storageAccounts/privateEndpointConnections/privateEndpoint.id'), '//'), '/')[2]]",
+                                                "notEquals": "[[subscription().subscriptionId]"
```

Comment:
This will enforce that no connection can be made _outside_ of the subscription scope of the resource, is that intentional?

Reply:
There is no way to check the tenant ID if the product team has not introduced new properties that we can validate against. They only way is to control this on a subscription-scope. Short answer: Yes, it is intentional.

Reply:
Since we cannot deterministically block across tenants, and the solution is to also prevent any cross-subscription scenarios, we should skip this for now please.

Reply:
Removed